kafka-tools/load-test/load.go

```diff
@@ -19,25 +20,38 @@ const (
 	serviceName          = "dis-search-upstream-stub"
 	KafkaTLSProtocolFlag = "TLS"
 	// Default values for message counts
-	defaultLegacyMessages = 6000
-	defaultNewMessages    = 100
+	defaultLegacySearchMessages  = 0
+	defaultSearchContentMessages = 500
 )
 
-func sendMessageToKafka(producer *kafka.Producer, item models.Resource, wg *sync.WaitGroup) {
+func makeStubTraceID(loopIndex int) string {
+	return fmt.Sprintf("stub-%d-%d", time.Now().UnixMilli(), loopIndex)
+}
+
+func sendMessageToKafka(producer *kafka.Producer, item models.Resource, loopIndex int, wg *sync.WaitGroup) {
 	defer wg.Done()
 
-	var messageBytes []byte
-	var err error
-	var eventType string
+	var (
+		messageBytes []byte
+		err          error
+		eventType    string
+		traceID      string
+	)
 
 	// Marshal the resource to Kafka message format based on the topic type
 	switch r := item.(type) {
 	case models.ContentUpdatedResource:
 		// Marshal for the legacy topic (ContentPublishedEvent)
+		if r.TraceID != "" {
+			traceID = makeStubTraceID(loopIndex)
+		}
```

Comment:
You've not put it back in the message to send. 
```suggestion
		if r.TraceID != "" {
			r.TraceID = makeStubTraceID(loopIndex)
		}
		traceID = r.TraceID
```